mayur/cart/shop.rb

```diff
@@ -0,0 +1,65 @@
+class Shop
+
+  attr_reader :products, :codes
+
+  def initialize(products = {}, codes = {})
+    @cart = {}
+    @products = products
+    @codes = codes
+  end
+
+  def update_inventory(products, codes)
+    @products = products
+    @codes = codes
+  end
+
+  def display()
+    puts @cart
+  end
+
+  def add_to_cart(name = nil,need = nil)
+    need = need.to_i
+    return @cart if (name.nil? || need.nil?)
+    index = @products.find_index { |p| p[:name] == name }
+    if !index.nil?
+      product = @products[index]
+      return @cart if need <= 0
+      if need > product[:quantity]
+        need = product[:quantity]
+        @products.delete_at(index)
+      else
+        @products.map do |x|
+          x[:quantity] - need if x[:name] == name
+        end
```

Comment:
May we should use map! instead of map because map will not change original object

Reply:
Done Thank you